sql/auth/sql_authorization.cc

```diff
@@ -1184,8 +1185,12 @@ void get_sp_access_map(
     */
 
     if (!strcmp(acl_user->user, user) &&
-        grant_proc->host.compare_hostname(acl_user->host.get_host(),
-                                          acl_user->host.get_host())) {
+        ((!effective_grants &&
```

Comment:
The condition (!effective_grants && !my_strcasecmp()) || (effective_grants && compare_hostname()) is repeated several times - please extract (and remove one of the "effective_grants" in effective_grants || !effective_grants)